src/EditorFeatures/Core/Implementation/CodeFixes/CodeFixService.cs

```diff
@@ -174,15 +174,21 @@ public async Task<ImmutableArray<CodeFixCollection>> GetFixesAsync(Document docu
 
                 cancellationToken.ThrowIfCancellationRequested();
 
-                aggregatedDiagnostics ??= new SortedDictionary<TextSpan, SortedSet<DiagnosticData>>();
-                aggregatedDiagnostics.GetOrAdd(diagnostic.GetTextSpan(), _ => new SortedSet<DiagnosticData>(s_diagnosticDataComparerById)).Add(diagnostic);
+                aggregatedDiagnostics ??= new SortedDictionary<TextSpan, List<DiagnosticData>>();
+                aggregatedDiagnostics.GetOrAdd(diagnostic.GetTextSpan(), _ => new List<DiagnosticData>()).Add(diagnostic);
             }
 
             if (aggregatedDiagnostics == null)
             {
                 return ImmutableArray<CodeFixCollection>.Empty;
             }
 
+            // Order diagnostics by DiagnosticId so the fixes are in a deterministic order.
+            foreach (var diagnosticsWithSpan in aggregatedDiagnostics.Values)
+            {
+                diagnosticsWithSpan.Sort(s_diagnosticDataComparisonById);
```

Comment:
❔ How often does this get called? Sorting with a `Comparison<T>` will cause an allocation with each pass (creating a `Comparer<T>` wrapping the `Comparison<T>`.

Reply:
This will be invoked once for each `Ctrl + .` invocation, so I think we should be fine.